projects/plugins/jetpack/extensions/shared/execution-lock.js

```diff
@@ -0,0 +1,74 @@
+/**
```

Comment:
Maybe this file/directory could do with some documentation with an example use?

Reply:
Do you need more examples besides the ones in the test file? Could you specify any example that you think could add value? I'll add any example requested as a test.

Reply:
As spoke in person, added a reference to the test file in this file.